crates/bevy_pbr/src/material.rs

```diff
@@ -232,6 +233,27 @@ where
     }
 }
 
+fn extract_material_meshes<M: Material>(
```

Comment:
I'm confused what the purpose of this system is. Do non-visible mesh handles not get extracted anymore?

Reply:
It inserts a default MaterialBindGroupId component at the same time. This avoids having to insert it as a new component later which then means all component data for all the entities have to be moved. I had noted that in a comment below. But I suppose the extract component function could return that. I’m replacing this with SparseSet in my next PR anyway.

Reply:
Yeah I got the MaterialBindGroupId part. I just wasn't sure what else it was doing, I didn't quite understand the code. Got it.

Reply:
I'm not going to change this, just because I know that I am changing it in my next PR to not be able to use the ExtractComponentPlugin anyway.